framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2763,30 +2763,12 @@ public ParameterizedExecutableType constructorFromUse(NewClassTree tree) {
             if (superCon.getParameterTypes().size() == con.getParameterTypes().size()) {
                 con.setParameterTypes(superCon.getParameterTypes());
             } else {
-                // If the super class of the anonymous class has an enclosing type, then it is the
-                // first parameter of the anonymous constructor. For example,
-                // class Outer { class Inner {} }
-                //  new Inner(){};
-                // Then javac creates the following constructor:
-                //  (.Outer x0) {
-                //   x0.super();
-                //   }
-                // So the code below deals with this.
-                List<AnnotatedTypeMirror> p =
-                        new ArrayList<>(superCon.getParameterTypes().size() + 1);
-                if (TreeUtils.hasSyntheticArgument(tree)) {
-                    p.add(con.getParameterTypes().get(0));
-                    con.setReceiverType(superCon.getReceiverType());
-                } else if (con.getReceiverType() != null) {
-                    // Because the anonymous constructor doesn't have annotated receiver type,
-                    // we copy the receiver type from the super constructor invoked in the anonymous
-                    // constructor and add it to the parameterTypes as the first element.
-                    con.setReceiverType(superCon.getReceiverType());
-                    p.add(con.getReceiverType());
-                } else {
-                    p.add(con.getParameterTypes().get(0));
-                }
-                p.addAll(1, superCon.getParameterTypes());
+                // Because the anonymous constructor doesn't have annotated receiver type,
```

Comment:
The deleted comment contained an example. Would that example still be useful explanation somewhere?

Reply:
Yes, I think it can be kept here, do you agree?

Reply:
Sure, as long as the example clarifies how things work now, keep it.